packages/e2e-test-utils/src/deactivate-plugin.js

```diff
@@ -13,6 +13,11 @@ import { visitAdminPage } from './visit-admin-page';
 export async function deactivatePlugin( slug ) {
 	await switchUserToAdmin();
 	await visitAdminPage( 'plugins.php' );
+	const deleteLink = await page.$( `tr[data-slug="${ slug }"] .delete a` );
+	if ( deleteLink ) {
+		await switchUserToTest();
```

Comment:
Alternatively, it could be more expressive for "common" logic as:

```js
export async function deactivatePlugin( slug ) {
	await switchUserToAdmin();
	await visitAdminPage( 'plugins.php' );
	const deleteLink = await page.$( `tr[data-slug="${ slug }"] .delete a` );
	if ( ! deleteLink ) {
		await page.click( `tr[data-slug="${ slug }"] .deactivate a` );
		await page.waitForSelector( `tr[data-slug="${ slug }"] .delete a` );
	}
	await switchUserToTest();
}

Reply:
That works too. No strong opinion about either way.